src/components/FilterDropdown.jsx

```diff
@@ -16,15 +18,22 @@ const useStyles = makeStyles((theme) => ({
 }));
 
 function FilterDropdown(props) {
-  const { label, items, defaultValue, removeNone, onChange } = props;
-  const [choice, setChoice] = useState(defaultValue);
+  const { label, items, value, removeNone, onChange } = props;
   const classes = useStyles();
+  const location = useLocation();
+  const params = queryString.parse(location.search, { parseNumbers: true });
+  const [choice, setChoice] = useState(
+    value === 'sorting'
+      ? params.sortBy === undefined
+        ? 'deadlineAsc'
+        : params.sortBy
+      : ''
+  );
```

Comment:
lint fails because of nested ternary expression, any suggestions on how to go about this?

Reply:
Nvm got it, I think can you check to ensure I did it correctly?